runtests.py

```diff
@@ -63,23 +63,71 @@
 
 assert all(cmd in cmds for cmd in FAST_FAIL)
 
-if not set(args).issubset(cmds):
-    print("usage:", prog, " ".join('[%s]' % k for k in cmds))
-    exit(1)
-
-if not args:
-    args = list(cmds)
-
-status = 0
 
-for arg in args:
-    cmd = cmds[arg]
-    print('run %s: %s' % (arg, cmd))
+def run_cmd(name: str) -> int:
+    status = 0
+    cmd = cmds[name]
+    print('run %s: %s' % (name, cmd))
     res = (system(cmd) & 0x7F00) >> 8
     if res:
-        print('\nFAILED: %s' % arg)
+        print('\nFAILED: %s' % name)
         status = res
-        if arg in FAST_FAIL:
+        if name in FAST_FAIL:
             exit(status)
+    return status
+
+
+def start_background_cmd(name: str) -> Popen:
+    cmd = cmds[name]
+    proc = subprocess.Popen(cmd,
+                            shell=True,
+                            stderr=subprocess.STDOUT,
+                            stdout=subprocess.PIPE)
+    return proc
+
+
+def wait_background_cmd(name: str, proc: Popen) -> int:
+    output = proc.communicate()[0]
+    status = proc.returncode
+    print('run %s: %s' % (name, cmds[name]))
+    if status:
+        print(output.decode().rstrip())
+        print('\nFAILED: %s' % name)
+        if name in FAST_FAIL:
+            exit(status)
+    return status
+
+
+def main() -> None:
+    prog, *args = argv
+
+    if not set(args).issubset(cmds):
+        print("usage:", prog, " ".join('[%s]' % k for k in cmds))
+        exit(1)
+
+    if not args:
+        args = list(cmds)
+
+    status = 0
+
+    if args[:2] == ['self', 'lint']:
```

Comment:
Maybe instead check if both self and lint are somewhere in list? Otherwise it would be sad to accidentally switch the order and get a slower run.